script/deploy/1_DeployOFT.s.sol

```diff
@@ -0,0 +1,140 @@
+/* solhint-disable no-console, gas-custom-errors */
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import {BaseScript} from "script/BaseScript.s.sol";
+
+import {L1YnOFTAdapterUpgradeable} from "@/L1YnOFTAdapterUpgradeable.sol";
+
+import {L2YnERC20Upgradeable} from "@/L2YnERC20Upgradeable.sol";
+import {L2YnOFTAdapterUpgradeable} from "@/L2YnOFTAdapterUpgradeable.sol";
+
+import {TimelockController} from "@openzeppelin/contracts/governance/TimelockController.sol";
+import {console} from "forge-std/console.sol";
+
+// forge script DeployOFT --rpc-url ${rpc}
+// --sig "run(string calldata)" ${path} \
+// --account ${deployerAccountName} --sender ${deployer} \
+// --broadcast --etherscan-api-key ${api} --verify
+
+contract DeployOFT is BaseScript {
+    function run(string calldata _jsonPath) public {
+        _loadInput(_jsonPath);
+
+        address deployer = msg.sender;
+
+        if (!isContract(currentDeployment.oftAdapterTimelock)) {
+            console.log("Deploying Timelock");
+            address admin = getData(block.chainid).PROXY_ADMIN;
+
+            address[] memory proposers = new address[](1);
+            proposers[0] = admin;
+
+            address[] memory executors = new address[](1);
+            executors[0] = admin;
+
+            uint256 minDelay = getMinDelay(block.chainid);
+
+            bytes memory constructorParams = abi.encode(minDelay, proposers, executors, admin);
+            bytes memory contractCode = abi.encodePacked(type(TimelockController).creationCode, constructorParams);
+
+            bytes32 timelockSalt = createTimelockSalt(deployer);
+            address predictedTimelock = CREATE3_FACTORY.getDeployed(deployer, timelockSalt);
+
+            vm.startBroadcast();
+            currentDeployment.oftAdapterTimelock = CREATE3_FACTORY.deploy(timelockSalt, contractCode);
+            vm.stopBroadcast();
+            console.log("Timelock deployed at: ", currentDeployment.oftAdapterTimelock);
+            require(predictedTimelock == currentDeployment.oftAdapterTimelock, "Timelock Prediction mismatch");
+        } else {
+            console.log("Already deployed Timelock at: ", currentDeployment.oftAdapterTimelock);
+        }
+
+        if (currentDeployment.isL1) {
```

Comment:
STEP 2: For L1, ERC20 is already deployed so we don't do anything. 
For L2, check if the ERC20 is already deployed, else deploy it using CREATE3